src/global/selectors/chats.ts

```diff
@@ -159,6 +160,55 @@ export function selectChatFolder<T extends GlobalState>(global: T, folderId: num
   return global.chatFolders.byId[folderId];
 }
 
+export function selectChatFolders<T extends GlobalState>(global: T, chat: ApiChat, user: ApiUser | undefined) {
```

Comment:
This method returns a new array instance each time, which will break memoization in Chat's `withGlobal` leading to redundant re-renders on every global change. 

Apart from that, the function itself is quite slow, as it has nested loops and it's itself called within a loop (for every Chat). I would suggest leveraging the `src/util/folderManager` module which is highly-optimized and is suited better for this case.